src/main/java/com/puppycrawl/tools/checkstyle/site/PropertiesMacro.java

```diff
@@ -165,13 +169,34 @@ private static void writeTablePropertiesRows(Sink sink)
         final Map<String, DetailNode> propertiesJavadocs = SiteUtil
                 .getPropertiesJavadocs(properties, currentModuleName, currentModuleFile);
 
-        for (String property : properties) {
+        final List<String> orderProperties = orderProperties(properties);
+
+        for (String property : orderProperties) {
             final DetailNode propertyJavadoc = propertiesJavadocs.get(property);
             final DetailNode currentModuleJavadoc = propertiesJavadocs.get(currentModuleName);
             writePropertyRow(sink, property, propertyJavadoc, instance, currentModuleJavadoc);
         }
     }
 
+    /**
+     * Reorder properties to always have the 'tokens' property last (if present).
+     *
+     * @param properties module properties.
+     * @return Collection of ordered properties.
+     *
+     */
+    private static List<String> orderProperties(Set<String> properties) {
+
+        final List<String> validProperties = new LinkedList<>(properties);
+
+        if (validProperties.remove(TOKENS_PROPERTY)) {
+            validProperties.add(TOKENS_PROPERTY);
+        }
+
+        return validProperties;
+
```

Comment:
Please return immutable list here, as suggested in other PR:
```suggestion
      if (properties.remove(TOKENS_PROPERTY)) {
            properties.add(TOKENS_PROPERTY);
        }
        return List.copyOf(properties);
```

Reply:
Rewrite the conditions, direclty removing from the set and adding the set causing an error, hence creating list first and then after performing the operation returning immutable list